src/Microsoft.ML.AutoML/Assembly.cs

```diff
@@ -3,11 +3,11 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Runtime.CompilerServices;
-using Microsoft.ML;
 
 [assembly: InternalsVisibleTo("Microsoft.ML.AutoML.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010015c01ae1f50e8cc09ba9eac9147cf8fd9fce2cfe9f8dce4f7301c4132ca9fb50ce8cbf1df4dc18dd4d210e4345c744ecb3365ed327efdbc52603faa5e21daa11234c8c4a73e51f03bf192544581ebe107adee3a34928e39d04e524a9ce729d5090bfd7dad9d10c722c0def9ccc08ff0a03790e48bcd1f9b6c476063e1966a1c4")]
 [assembly: InternalsVisibleTo("mlnet, PublicKey=00240000048000009400000006020000002400005253413100040000010001004b86c4cb78549b34bab61a3b1800e23bfeb5b3ec390074041536a7e3cbd97f5f04cf0f857155a8928eaa29ebfd11cfbbad3ba70efea7bda3226c6a8d370a4cd303f714486b6ebc225985a638471e6ef571cc92a4613c00b8fa65d61ccee0cbe5f36330c9a01f4183559f1bef24cc2917c6d913e3a541333a1d05d9bed22b38cb")]
 [assembly: InternalsVisibleTo("mlnet.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010015c01ae1f50e8cc09ba9eac9147cf8fd9fce2cfe9f8dce4f7301c4132ca9fb50ce8cbf1df4dc18dd4d210e4345c744ecb3365ed327efdbc52603faa5e21daa11234c8c4a73e51f03bf192544581ebe107adee3a34928e39d04e524a9ce729d5090bfd7dad9d10c722c0def9ccc08ff0a03790e48bcd1f9b6c476063e1966a1c4")]
 [assembly: InternalsVisibleTo("Benchmark, PublicKey=00240000048000009400000006020000002400005253413100040000010001004b86c4cb78549b34bab61a3b1800e23bfeb5b3ec390074041536a7e3cbd97f5f04cf0f857155a8928eaa29ebfd11cfbbad3ba70efea7bda3226c6a8d370a4cd303f714486b6ebc225985a638471e6ef571cc92a4613c00b8fa65d61ccee0cbe5f36330c9a01f4183559f1bef24cc2917c6d913e3a541333a1d05d9bed22b38cb")]
-
-[assembly: WantsToBeBestFriends]
+[assembly: InternalsVisibleTo("Microsoft.ML.CodeGenerator, PublicKey=00240000048000009400000006020000002400005253413100040000010001004b86c4cb78549b34bab61a3b1800e23bfeb5b3ec390074041536a7e3cbd97f5f04cf0f857155a8928eaa29ebfd11cfbbad3ba70efea7bda3226c6a8d370a4cd303f714486b6ebc225985a638471e6ef571cc92a4613c00b8fa65d61ccee0cbe5f36330c9a01f4183559f1bef24cc2917c6d913e3a541333a1d05d9bed22b38cb")]
+[assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
```

Comment:
I'm not sure I understand these two `InternalsVisibleTo` attributes:

```
[assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder
[assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder.AutoMLService
```

Why are they required for this PR?

Reply:
You're right, they are not needed for this PR and not part of subtask for moving CodeGen to master, 
To keep changes more granular we could add the two in a separate PR. will remove in next commit

cc: @LittleLittleCloud

Reply:
> I'm not sure I understand these two `InternalsVisibleTo` attributes:
> 
> ```
> [assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder
> [assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder.AutoMLService
> ```
> 
> Why are they required for this PR?

Because both `ModelBuilder` and `ModelBuilder.AutoMLService` require accessibility to `Pipeline` class ( in order to support try-it-out experience ) currently, which is internal

Reply:
Why does ModelBuilder need access to the `Pipeline` class? It should just get the `BestModel` and work with it.

Reply:
My opinion is that this change belongs in a separate PR. Sneaking a change like that into a "Code Gen" PR doesn't make sense.

Reply:
To generate project, trained pipeline need to be used as input to `CodeGen`

Reply:
If that is the case, then why is the `CodeGenerator` class public? It seems like it will be impossible for anyone to use it if they don't have access to this `Pipeline` class.

Reply:
The InternalsVisibleTo should be in sync with what is in features/automl branch, plus an extra item for Microsoft.ML.ModelBuilder.AutoMLService and one for the CodeGenerator test project added in this PR.

Refer to what's already included in features/AutoML below:

- https://github.com/dotnet/machinelearning/blob/features/automl/src/Microsoft.ML.AutoML/Assembly.cs
- https://github.com/dotnet/machinelearning/blob/features/automl/src/Microsoft.ML.CodeGen/Assembly.cs